packages/rocketchat-livechat/client/views/app/tabbar/visitorInfo.js

```diff
@@ -243,18 +250,17 @@ Template.visitorInfo.onCreated(function() {
 
 	if (currentData && currentData.rid) {
 		this.autorun(() => {
-			const room = ChatRoom.findOne(currentData.rid);
-			if (room && room.v && room.v._id) {
-				this.visitorId.set(room.v._id);
-			} else {
-				this.visitorId.set();
-			}
+			const room = Rooms.findOne(currentData.rid);
+			this.visitorId.set(room && room.v && room.v._id);
+			this.departmentId.set(room && room.departmentId);
 		});
 
 		this.subscribe('livechat:visitorInfo', { rid: currentData.rid });
+		this.subscribe('livechat:departments');
```

Comment:
you don't need to subscribe for ALL departments and filter only on client. you can pass the department id to `livechat:departments` and subscribe only the department you need

Reply:
Fixed.